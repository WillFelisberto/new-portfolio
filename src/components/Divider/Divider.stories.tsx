import { Meta, StoryObj } from '@storybook/react';
import Divider from '.';

const meta: Meta<typeof Divider> = {
  title: 'Components/Divider',
  component: Divider,
  parameters: {
    layout: 'fullscreen',
  },
  args: {
    className: '',
  },
};

export default meta;

type Story = StoryObj<typeof Divider>;

export const Default: Story = {};

export const WithCustomStyles: Story = {
  args: {
    className: 'border-b-red-500', // Exemplo de personalização
  },
};
