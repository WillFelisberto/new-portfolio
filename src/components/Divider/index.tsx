import { cn } from '@/lib/utils';

type DividerProps = {
  className?: string;
};
const Divider = ({ className }: DividerProps) => {
  return (
    <div className={cn('my-8 w-full border-b border-b-gray-800', className)} />
  );
};

export default Divider;
