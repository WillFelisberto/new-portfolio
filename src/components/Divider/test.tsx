import { render } from '@testing-library/react';

import Divider from '.';

describe('<Divider />', () => {
  // Renders divider with default styling classes
  it('should render with default styling classes when no className prop is provided', () => {
    const { container } = render(<Divider />);

    const divider = container.firstChild;

    expect(divider).toHaveClass(
      'my-8',
      'w-full',
      'border-b',
      'border-b-gray-800',
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly when className prop is undefined', () => {
    const { container } = render(<Divider className={undefined} />);

    const divider = container.firstChild;

    expect(divider).toHaveClass(
      'my-8',
      'w-full',
      'border-b',
      'border-b-gray-800',
    );
    expect(divider).not.toHaveClass('undefined');
  });
});
