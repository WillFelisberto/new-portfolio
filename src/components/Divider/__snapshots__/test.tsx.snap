// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> should render with default styling classes when no className prop is provided 1`] = `
<div
  class="my-8 w-full border-b border-b-gray-800"
/>
`;
